### PR TITLE
feat(cactus-cmd-gui-app): extend common cacti gui app to operate with  fabric hyperledger

### PR DESCRIPTION
**Pull request by barnapa**
Add feature and options to switch in GUI app to visualize fabric hyperledger ledger data stored in supabase by persistence plugins

Depends on: https://github.com/hyperledger/cacti/pull/2331

Signed-off-by: Barnaba Pawelczak [barnaba.pawelczak@fujitsu.com](mailto:barnaba.pawelczak@fujitsu.com)

### CODE REVIEW

File: tsconfig.json
Comment (petermetz) [inline tsconfig.json:106]: @barnapa The GUI changes are looking good, but may I ask why did you remove the `packages/cactus-test-verifier-client/tsconfig.json` file from the typescript compilation altogether?

Comment (barnapa) [reply tsconfig.json]: ok added

**Comment by ryjones** [comment]
@petermetz I've fixed the commit message errors, squashed, and rebased.

**Comment by petermetz** [comment]
> @petermetz I've fixed the commit message errors, squashed, and rebased.

@ryjones Noo, I wanted the separate commits so that we can filter the diff based on what's in this PR and what's in the the two parent PRs (the ones that this depends on getting merged first).

It's fine, I will just wait for the parents to get merged and then a squash+rebase will sort everything out, e.g. we won't see the combined diff of all 3 PRs in a single commit just the diff that actually belongs to this PR.

**Comment by ryjones** [comment]
I'll undo it if I can

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
I think we should maintain the same package folder structure as other packages.
Moving the content of `src/` to `src/main/language_name/` should fix this.

**Comment by petermetz** [comment]
> I think we should maintain the same package folder structure as other packages. Moving the content of `src/` to `src/main/language_name/` should fix this.

+1 on this. I'll wait with the detailed review until the parent PRs get merged and we can untangle the diff.

**Review by petermetz — CHANGES_REQUESTED** [review]
@barnapa Please re-request review once the parent's have been merged and this one had been rebased onto `upstream/main` afterwards.

**Review by petermetz — CHANGES_REQUESTED** [review]
> you prefer to have css all files in separate files in src/main/css and typescript and tsx files in src/main/typescript/ like src/main/typescript/pages/Accounts/ and we can keep the subfolders structure corresponding to pages of website project ? that would be ok ? @ryjones

@barnapa It would be preferred, BUT I also don't want to make it hard for existing react devs to contribute so I'm gonna change my vote on this and say let's keep that structure in place as is.

Other things that you can do to actually integrate with the monorepo:
1. build is what I said on discord about the build script (e.g. running yarn build:dev:frontend ends up building the GUI packages)
2. rename from cactus to cacti (everything within your diff)
3. don't commit a package-lock.json - we use yarn and the lock file for it is in the project root
4. don't have duplicate dependencies/dev-dependencies, e.g. ones that are already provided by the monorepo root package.json file don't need to be re-declared here in the package itself unless there's a specific need for a different version that cannot be satisfied
5. make the tsconfig.json file consistent with the rest of the tsconfig.json files, for example see the cactus-common package's tsconfig.json file. I don't know if this is another thing that just breaks down due to some hardcoded thing about react but if that is the case do let me know. I'm trying hard not to bikeshed everything but also not let things diverge needlessly.

**Comment by petermetz** [comment]
Suggested edit:
```diff
diff --git a/lerna.json b/lerna.json
index 37162da58..d66a7a409 100644
--- a/lerna.json
+++ b/lerna.json
@@ -2,7 +2,10 @@
   "packages": [
     "packages/cactus-*",
     "examples/cactus-*",
-    "extensions/cactus-*"
+    "extensions/cactus-*",
+    "packages/cacti-*",
+    "examples/cacti-*",
+    "extensions/cacti-*"
   ],
   "version": "1.2.0",
   "npmClient": "yarn",
diff --git a/package.json b/package.json
index 54868313b..2787766f5 100644
--- a/package.json
+++ b/package.json
@@ -6,7 +6,10 @@
     "packages": [
       "packages/cactus-*",
       "examples/cactus-*",
-      "extensions/cactus-*"
+      "extensions/cactus-*",
+      "packages/cacti-*",
+      "examples/cacti-*",
+      "extensions/cacti-*"
     ],
     "nohoist": [
       "**/iroha-helpers"
@@ -42,7 +45,7 @@
     "build:prod:frontend": "lerna run build:prod:frontend",
     "build:dev": "npm-run-all build:dev:backend webpack:dev:web build:dev:frontend",
     "build:dev:backend": "npm-run-all lint clean codegen tsc build:dev:backend:postbuild",
-    "build:dev:frontend": "lerna run build:dev:frontend --scope=\"@hyperledger/cactus-example-supply-chain-frontend\" && lerna run build:dev:frontend --scope=\"@hyperledger/cactus-example-carbon-accounting-frontend\"&& lerna run build:dev:frontend --scope=\"packages/cacti-cmd-gui-app\"",
+    "build:dev:frontend": "lerna run build:dev:frontend --scope=\"@hyperledger/cactus-example-supply-chain-frontend\" && lerna run build:dev:frontend --scope=\"@hyperledger/cactus-example-carbon-accounting-frontend\"&& lerna run build:dev:frontend --scope=\"@hyperledger/cacti-cmd-gui-app\"",
     "build:dev:common": "lerna exec --stream --scope '*/*common' -- 'del-cli dist/** && tsc --project ./tsconfig.json && webpack --env=dev --target=node --config ../../webpack.config.js'",
     "build:dev:backend:postbuild": "lerna run build:dev:backend:postbuild",
     "test:cmd-api-server": "tap --ts --timeout=600 \"packages/cactus-*cmd-api-server/src/test/typescript/{unit,integration}/\"",
diff --git a/packages/cacti-cmd-gui-app/package.json b/packages/cacti-cmd-gui-app/package.json
index 4bb828201..4eab4f134 100644
--- a/packages/cacti-cmd-gui-app/package.json
+++ b/packages/cacti-cmd-gui-app/package.json
@@ -1,6 +1,6 @@
 {
-  "name": "@hyperledger/cactus-cmd-gui-app",
-  "version": "1.1.3",
+  "name": "@hyperledger/cacti-cmd-gui-app",
+  "version": "2.0.0-alpha.1",
   "description": "Cacti GUI for visualizing ledger data.",
   "keywords": [
     "Hyperledger",
@@ -44,6 +44,7 @@
     "start": "vite",
     "serve": "vite preview",
     "build": "yarn run build:prod:frontend",
+    "build:dev:frontend": "vite build --mode=development",
     "build:prod:frontend": "vite build"
   },
   "devDependencies": {
diff --git a/tsconfig.json b/tsconfig.json
index 9599f1f59..2e73ef78e 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -5,13 +5,13 @@
       "path": "./packages/cactus-api-client/tsconfig.json"
     },
     {
-      "path": "./packages/cacti-cmd-api-server/tsconfig.json"
+      "path": "./packages/cactus-cmd-api-server/tsconfig.json"
     },
     {
       "path": "./packages/cactus-cmd-socketio-server/tsconfig.json"
     },
     {
-      "path": "./packages/cactus-cmd-gui-app/tsconfig.json"
+      "path": "./packages/cacti-cmd-gui-app/tsconfig.json"
     },
     {
       "path": "./packages/cactus-common/tsconfig.json"

```

**Review by petermetz — CHANGES_REQUESTED** [review]
@barnapa I added some suggestions to fix the build. They might need some conflict resolution later on (versions and the project-common files like lerna.json and root package.json) but at least you can use it as a reference later on.

**Review by petermetz — CHANGES_REQUESTED** [review]
@barnapa LGTM now just please do the usual chores to clean it up and get it ready for merge: squash, rebase, fix the DCO, commit lint checks, etc. If you need any help with any of that just let me know!

**Review by petermetz — APPROVED** [review]
@barnapa LGTM, thank you!
The CI is flaky at the moment so it might take a few retries but we should be able to merge it soon (assuming other reviews are on the way because at the moment the one I'm typing is the only one)

**Comment by barnapa** [comment]
> I think we should maintain the same package folder structure as other packages. Moving the content of `src/` to `src/main/language_name/` should fix this.

not really we discuss this matter for frontend few topics before.

**Comment by barnapa** [comment]
> @barnapa LGTM, thank you! The CI is flaky at the moment so it might take a few retries but we should be able to merge it soon (assuming other reviews are on the way because at the moment the one I'm typing is the only one)

@izuru0 @takeutak  please check this one

**Review by izuru0 — APPROVED** [review]
LGTM

**Comment by barnapa** [comment]
> I think we should maintain the same package folder structure as other packages. Moving the content of `src/` to `src/main/language_name/` should fix this.
Dear
@jagpreetsinghsasan Could you approve as we discuss this matter and conclusion was to keep actual structure for frontend

**Review by jagpreetsinghsasan — APPROVED** [review]

**Comment by barnapa** [comment]
@petermetz Dear Peter can we now merge ?

**Comment by petermetz** [comment]
> @petermetz Dear Peter can we now merge ?

@barnapa Yeah we are good to go from my side. Please rebase onto upstream/main (and resolve the merge conflict via that) and then I should be able to take care of the rest.

**Comment by petermetz** [comment]
@barnapa Sorry, a little more delay because I can't approve it due to @jagpreetsinghsasan already having approved it and we both work for the same organization. We need someone else (any other maintainer) reviewing it from any of the two other orgs (@izuru0 @VRamakrishna @sandeepnRES @takeutak )

**Review by izuru0 — APPROVED** [review]
LGTM

**Pull request merged**
